### ISSUE TITLE
Race condition when triggering github sync from openfn while deploy actions are running

### ISSUE THREAD

**Issue by midigofrank** [OpenFn/lightning#2179]
Will add more details later..

@mtuchi ran into a situation where one of his deploy workflows could not complete because his branch was out of date with the remote branch. Know that git error which tells you to pull from the remote first?

He said he did a push to his branch and then triggered a sync from openfn. My gut tells me that the sync from lightning did a commit before one of his workflows could complete. I think this is possible given that the `deploy` workflows are currently running sequentially, while the `pull` ones aren't.

Maybe we should have the `pull` actions running in the `openfn-deployment` group as well so that we avoid all these race conditions.

**Comment by midigofrank** [comment]
@taylordowns2000 thoughts?

**Comment by taylordowns2000** [comment]
Makes sense. I didn't realize that our fix didn't put all these workflows in the same group. Please raise a PR. I see no harm in forcing all the OpenFn actions to be sequential.

**Comment by christad92** [comment]
@midigofrank did we fix this?

**Comment by midigofrank** [comment]
@christad92 nope.

**Comment by midigofrank** [comment]
Oops!! This isn't as straight forward as I thought
https://taurit.pl/github-canceling-since-a-higher-priority-waiting-request-exists/
